src/bp/ui-admin/src/Pages/Server/BotHealth.tsx

```diff
@@ -0,0 +1,135 @@
+import { Button, Tooltip } from '@blueprintjs/core'
+import { ServerHealth } from 'common/typings'
+import _ from 'lodash'
+import React, { FC, useEffect, useState } from 'react'
+import { connect } from 'react-redux'
+import ReactTable from 'react-table'
+import api from '~/api'
+import { fetchBotHealth } from '~/reducers/bots'
+import { toastFailure, toastSuccess } from '~/utils/toaster'
+import confirmDialog from '~/App/ConfirmDialog'
+
+interface Props {
+  health: ServerHealth[]
+  fetchBotHealth: () => void
+}
+
+const getKey = entry => `${entry.hostname} (${entry.serverId})`
+
+const BotHealth: FC<Props> = props => {
+  const [data, setData] = useState<any>()
+  const [columns, setColumns] = useState<any>([])
+
+  useEffect(() => {
+    if (!props.health) {
```

Comment:
```suggestion
    if (props.length === 0) {
```
because a empty array is truthy:
```
> ![]
false
```

Reply:
I'll fix the typings, props.health is undefined by default